### PR TITLE
Konvoy

### PR DESCRIPTION
**Pull request by karl-cardenas-coding**
## Describe the Change

This PR adds instructions on creating a custom RHEL image with Konvoy for a VMware deployment.

- [RHEL and Konvoy](https://65e275cfffad9246f2180163--docs-spectrocloud.netlify.app/byoos/usecases/vmware/konvoy)
- [Konvoy Pack Page](https://65e275cfffad9246f2180163--docs-spectrocloud.netlify.app/integrations/konvoy) 

## Review Changes

💻 [Preview URL](https://65e275cfffad9246f2180163--docs-spectrocloud.netlify.app/byoos/usecases/vmware/konvoy)

🎫 [PCP-2510](https://spectrocloud.atlassian.net/browse/PCP-2510)

[PCP-2510]: https://spectrocloud.atlassian.net/browse/PCP-2510?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: docs/docs-content/byoos/usecases/vmware/konvoy.md [hunk 1]
```diff
@@ -0,0 +1,465 @@
...
+
+    ```shell
+    mkdir --parents ~/builder
+    cd ~/builder
+    ```
+
+2.  Download the Konvoy builder release artifact from the GitHub releases page.
+
+    ```shell
+    wget https://github.com/mesosphere/konvoy-image-builder/releases/download/v2.8.5/konvoy-image-bundle-v2.8.5_linux_amd64.tar.gz
```
Comment (addetz) [inline docs/docs-content/byoos/usecases/vmware/konvoy.md:104]: Should we add a little comment about the versioning in this link? `We will be using version 2.8.5 in this guide`

File: docs/docs-content/byoos/usecases/vmware/konvoy.md [hunk 2]
```diff
@@ -0,0 +1,465 @@
...
+    ```
+
+3.  Extract the downloaded tarball and navigate to the builder directory.
+
+    ```shell
+    tar --extract --file  konvoy-image-bundle-v2.8.5_linux_amd64.tar.gz --directory builder && \
+    cd builder
+    ```
+
+4.  Export your vSphere and RHEL credentials as environment variables. Replace the placeholders with your credentials.
```
Comment (addetz) [inline docs/docs-content/byoos/usecases/vmware/konvoy.md:114]:
```suggestion
4.  Export your vSphere and RHEL credentials as environment variables. Replace the placeholders with your credentials. Execute the following commands to set these values in your terminal session.
```

File: docs/docs-content/byoos/usecases/vmware/konvoy.md [hunk 3]
```diff
@@ -0,0 +1,465 @@
...
+    the `ansible.builtin.reboot` task.
+
+        ```yaml
+        ansible.builtin.reboot:
+          post_reboot_delay: 120
+          pre_reboot_delay: 30
+          connect_timeout: 120
+        ```
+
+    {/* prettier-ignore */}
```
Comment (addetz) [inline docs/docs-content/byoos/usecases/vmware/konvoy.md:361]: What does this do?

File: docs/docs-content/byoos/usecases/vmware/konvoy.md [hunk 4]
```diff
@@ -0,0 +1,465 @@
...
+      use_sudo: false
+      NOPASSWD: ALL
+    # ssh_private_key_file = "" # can be exported as environment variable 'SSH_PRIVATE_KEY_FILE'
+    # ssh_agent_auth: false  # is set to true, ssh_password and ssh_private_key will be ignored
+    ```
+
+    </details>
+
+7.  Review the **images/common.yaml** file and modify the file if necessary. The **common.yaml** file is where you
+    specify the Kubernetes version. Refer to the[Konvoy](../../../integrations/konvoy.md) pack reference page to learn
```
Comment (addetz) [inline docs/docs-content/byoos/usecases/vmware/konvoy.md:189]:
```suggestion
    specify the Kubernetes version. Refer to the [Konvoy](../../../integrations/konvoy.md) pack reference page to learn
```

File: docs/docs-content/byoos/usecases/vmware/konvoy.md [hunk 5]
```diff
@@ -0,0 +1,465 @@
...
+1. Log in to the vSphere web client and navigate to the datastore folder where the custom RHEL with Konvoy image is
+   located. Verify that the image is present in the folder.
+
+2. Create a cluster profile and use the BYOOS pack to reference the custom RHEL with Konvoy image. Specify the path to
+   the custom RHEL with Konvoy image in the pack's YAML file. Refer to
+   [Usage with Cluster Profile](#usage-with-cluster-profile) section for more details.
+
+3. Deploy a cluster using the cluster profile you created.
+
+## Usage with Cluster Profile
```
Comment (addetz) [inline docs/docs-content/byoos/usecases/vmware/konvoy.md:441]: Should this section not have its own separate validation steps?

File: docs/docs-content/integrations/konvoy.md [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+description: "Konvoy Kubernetes distribution"
+hide_table_of_contents: true
+type: "integration"
+category: ["kubernetes", "amd64"]
+sidebar_class_name: "hide-from-sidebar"
+logoUrl: "https://registry.spectrocloud.com/v1/k8s-dashboard/blobs/sha256:2de5d88b2573af42d4cc269dff75744c4174ce47cbbeed5445e51a2edd8b7429?type=image/png"
+tags: ["packs", "konvoy", "kubernetes"]
+---
+
+[Konvoy](https://d2iq.com/products/konvoy) is a Kubernetes distribution by D2iQ. Konvoy is built on pure open-source
```
Comment (addetz) [inline docs/docs-content/integrations/konvoy.md:13]:
```suggestion
[Konvoy](https://d2iq.com/products/konvoy) is a Kubernetes distribution by D2iQ. It is built on pure open-source
```

File: docs/docs-content/integrations/konvoy.md [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+Kubernetes and includes a full suite of integrated and supported best-of-breed Day 2 platform applications for an
```
Comment (addetz) [inline docs/docs-content/integrations/konvoy.md:14]:
```suggestion
Kubernetes and includes a full suite of integrated and supported best-of-breed Day-2 platform applications for an
```

File: docs/docs-content/integrations/konvoy.md [hunk 8]
```diff
@@ -0,0 +1,116 @@
...
+
+- Operating System (OS) dependencies as listed in the table.
+
+### Supported Infrastructure Platforms
+
+Palette supports the following infrastructure platforms and Operating Systems (OS) combinations for Konvoy.
+
+| Infrastructure Platform | OS                       | Supported?         |
+| ----------------------- | ------------------------ | ------------------ |
+| VMware vSphere          | Red Hat Linux Enterprise | :white_check_mark: |
```
Comment (addetz) [inline docs/docs-content/integrations/konvoy.md:44]: Do we need this table while it has 1 row?

File: docs/docs-content/integrations/konvoy.md [hunk 9]
```diff
@@ -0,0 +1,116 @@
...
+
+data "spectrocloud_pack_simple" "k8s" {
+  name    = "kubernetes-konvoy"
+  version = "1.27.6"
+  type = "helm"
+  registry_uid = data.spectrocloud_registry.public_registry.id
+}
+```
+
+## References
```
Comment (addetz) [inline docs/docs-content/integrations/konvoy.md:110]: Should this guide not have a Validate?

**Review by addetz — APPROVED** [review]
Small-ish comments and questions. 🚀

File: docs/docs-content/byoos/usecases/vmware/konvoy.md [hunk 10]
```diff
@@ -0,0 +1,465 @@
...
+---
+
+This guide provides instructions for building a custom image using Red Hat Linux Enterprise (RHEL) with Konvoy. The
+guide assumes that you have a basic understanding of VMware vSphere, RHEL and [Konvoy](../../../integrations/konvoy.md).
+
+You will use the [Konvoy image builder](https://github.com/mesosphere/konvoy-image-builder) project to build the custom
+RHEL image with Konvoy. The custom image will be used to deploy a Kubernetes cluster on VMware vSphere.
+
+Carefully review the prerequisites and follow the steps in the order provided to build and use a custom RHEL with
+Konvoy. Make sure you have met all the prerequisites before you begin.
```
[the remark this answers is not in the document]
Comment (karl-cardenas-coding) [reply docs/docs-content/byoos/usecases/vmware/konvoy.md:19]: Good call.

File: docs/docs-content/byoos/usecases/vmware/konvoy.md (hunk 1 above)
Comment (karl-cardenas-coding) [reply docs/docs-content/byoos/usecases/vmware/konvoy.md:104]: ### Great suggestion

File: docs/docs-content/byoos/usecases/vmware/konvoy.md (hunk 3 above)
Comment (karl-cardenas-coding) [reply docs/docs-content/byoos/usecases/vmware/konvoy.md:361]: Oh man, @addetz prettier kept refusing to let the current indentation work, which is the only way to achieve the correct indentation. Prettier kept pushing the `<summary> element to the far left and breaking the build. The comment above tells prettier to backoff.

File: docs/docs-content/byoos/usecases/vmware/konvoy.md (hunk 5 above)
Comment (karl-cardenas-coding) [reply docs/docs-content/byoos/usecases/vmware/konvoy.md:441]: I didn't want to create a how-to as we already have how-tos that explain how to create a cluster profile. So I wrote something that speaks to the flow at a high level.

File: docs/docs-content/integrations/konvoy.md (hunk 8 above)
Comment (karl-cardenas-coding) [reply docs/docs-content/integrations/konvoy.md:44]: Yeah, we should comment out the table for now. More versions and combinations will be added but you're right.

**Review by tiwarisumit3 — APPROVED** [review]

**Pull request merged**